src/screens/ProfileScreen/index.js

```diff
@@ -242,6 +244,11 @@ const ProfileScreen = ({route}) => {
         description: null
       }));
       setInitialFollowerData(newData);
+
+      const followedUsers = newData.filter((item) => item.user.following);
+      const unfollowedUsers = newData.filter((item) => !item.user.following);
+      DiscoveryAction.setNewFollowedUsers(followedUsers, discoveryDispatch);
+      DiscoveryAction.setNewUnfollowedUsers(unfollowedUsers, discoveryDispatch);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here seems to be correct, but it might be more efficient to iterate over `newData` only once instead of twice (once for `followedUsers` and once for `unfollowedUsers`). You can use a single `forEach` loop to separate the followed and unfollowed users. Here's how you could do it:

```diff
- const followedUsers = newData.filter((item) => item.user.following);
- const unfollowedUsers = newData.filter((item) => !item.user.following);
+ const followedUsers = [];
+ const unfollowedUsers = [];
+ newData.forEach((item) => {
+   if (item.user.following) {
+     followedUsers.push(item);
+   } else {
+     unfollowedUsers.push(item);
+   }
+ });
```
This way, you only need to iterate over `newData` once, which should improve performance, especially when `newData` is large.


<!-- This is an auto-generated comment by OSS CodeRabbit -->